### PR TITLE
Fixes #4275 - Add comment to parallel betweenness example

### LINKED ISSUE #1480: Write a CONTRIBUTING file
Suggested in https://github.com/networkx/networkx/issues/1428, but I feel like it needs a separate issue.

It would be good if a maintainer or a core developer could take the time to write a proper `CONTRIBUTING` file that explains
- the desired code style;
- the preferred choice of python statements when multiple are available (e.g. https://github.com/networkx/networkx/issues/1479);
- the proper way of implementing core that do not come with classes (e.g. partitions);
- the desired code organization (what goes where);
- the current main development targets.

And probably other things that I can't come up with just now.

### LINKED ISSUE #4275: Parallel betweenness example bad runtimes in docs
There is a nice example in the gallery for [computing betweenness centrality with multiprocessing](https://networkx.org/documentation/stable/auto_examples/advanced/plot_parallel_betweenness.html). Unfortunately, the example in the docs show poorer performance for the parallel version than the non-parallelized version, likely due to the CI service only using single-core (or otherwise not very performant) nodes for computation. At the very least it might be worth adding some text somewhere to the example to explain why the example shows performance results that differ from what is expected.

It's also probably worth taking a look at the example in its entirety and consider alternative approaches to multiprocessing, e.g. `concurrent.futures`.

### PR DESCRIPTION
**Pull request by korbonits**
# Description

Add comment about single core CI/CD showing slower parallel output in example script.

Added text:

```
Note: The example output below shows that the non-parallel implementation is
faster. This is a limitation of our CI/CD pipeline running on a single core.

Depending on your setup, you may observe a 4x speed improvement.
```

Following the discussion in the ticket, I left the multiprocessing as-is because I think it probably makes MORE sense than using concurrent.futures.

Here was my output running the script:

```
⇒  python examples/algorithms/plot_parallel_betweenness.py

Computing betweenness centrality for:
Name:
Type: Graph
Number of nodes: 1000
Number of edges: 2991
Average degree:   5.9820
	Parallel version
		Time: 0.7211 seconds
		Betweenness centrality for node 0: 0.00201
	Non-Parallel version
		Time: 3.0109 seconds
		Betweenness centrality for node 0: 0.00201

Computing betweenness centrality for:
Name:
Type: Graph
Number of nodes: 1000
Number of edges: 5018
Average degree:  10.0360
	Parallel version
		Time: 0.8089 seconds
		Betweenness centrality for node 0: 0.00283
	Non-Parallel version
		Time: 3.8002 seconds
		Betweenness centrality for node 0: 0.00283

Computing betweenness centrality for:
Name:
Type: Graph
Number of nodes: 1000
Number of edges: 2000
Average degree:   4.0000
	Parallel version
		Time: 0.6724 seconds
		Betweenness centrality for node 0: 0.02636
	Non-Parallel version
		Time: 2.6978 seconds
		Betweenness centrality for node 0: 0.02636
```

### CODE REVIEW

**Comment by dschult** [comment]
Did you mean #1480 or was the title supposed to reference #4275

**Comment by korbonits** [comment]
@dschult the latter -- typo on my side -- thanks!

File: examples/algorithms/plot_parallel_betweenness.py [hunk 1]
```diff
@@ -10,8 +10,12 @@
 the contribution of those nodes to the betweenness centrality of the whole
 network. Here we divide the network in chunks of nodes and we compute their
 contribution to the betweenness centrality of the whole network.
-"""
 
+Note: The example output below shows that the non-parallel implementation is
+faster. This is a limitation of our CI/CD pipeline running on a single core.
+
+Depending on your setup, you may observe a 4x speed improvement.
```
Comment (dschult) [inline examples/algorithms/plot_parallel_betweenness.py:17]: ```you may observe a 4x speed improvement.``` 
is both specific (x4) and qualified (you may observe).  Which feels strange.
What do you think about:   
```Depending on your setup, you will likely observe a speedup. We seem to get a 4x improvement.```

Really, I'm fine either way -- or another way.  Thoughts?

Comment (rossbar) [reply examples/algorithms/plot_parallel_betweenness.py:17]: Agreed - I don't think we should get too specific. Ultimately the performance will depend on the hardware (and even more complicating factors like resource limitations for people running in containerized environments), so I'd keep the wording as generic as possible --- maybe @dschult's second suggestion (potentially even dropping the "We seem to get..." part).

Comment (korbonits) [reply examples/algorithms/plot_parallel_betweenness.py]: Thanks! Apologies for delay in responding, dealing with family emergency. Might consider adding `concurrent.futures` example in future PR -- looking forward to collaborating further

**Pull request merged**